dsl/pace/dsl/dace/doc_primer_orchestration.md

```diff
@@ -0,0 +1,101 @@
+
+Build
+-----
+
+Orchestrated code won't build the same way the gt backend builds. The build pipeline will lead to a single folder with code & `.so`. In the case of the driver main call, this would be in `.gt_cache_*/dacecache/pace_driver_driver_Driver__critical_path_step_all`.
+
+Code goes through phases before being ready to execute:
+* stencils are `parsed` into non-expanded SDFG (gt4py takes care of this),
+* all code is `parsed` into a single SDFG with stencils' SDFG included (dace takes care of this and the following steps),
+* a first `simplify` is applied to the SDFG to optimize the memory flow,
+* we apply the custom `splittable_region_expansion` which optimize small regions (_major_ speed up),
+* `expand` will expand all the stencils to a fully workable SDFG (with tasklet filled)
+* another `simplify` is applied,
+* the memory that can is flagged to be `pooled`,
+* `code generation` into a single file for CPU or two for GPU (a `.cpp` and a `.cu`),
+* the SDFG is analysed for memory consumption.
+
+
+Orchestration comes with it's own distributed compilation (could be merged with gt). It compiles the top tile and distriubutes the results to other ranks. This uses a couple of hypothesis that limits how to build/execute. The major one is that any decomposition from `(3,3)` upward will require the following workflow:
+- compile on `(3,3)`,
+- copy caches 0 to 8 (top tile) to target decomposition run dir,
+- execute (`FV3_DACEMODE=Run`) target decompoposition.
+
+The other limitations is that it only is protecting bad runs when resolution X == resolution Y.
+
+The distributed compilation can be deactivated in `dace_config.py` by turning `DEACTIVATE_DISTRIBUTED_DACE_COMPILE` to `True`.
+
+Execution
+---------
+
+If orchestration gives us the best speed, remember that it execute _one single .so_ and therefore cannot be interrupted with regular Python debugging techniques.
+
+Pitfalls & remedies
+-------------------
+
+_Callback_
+
+Escape DaCe by decorating a function with `@dace_inibitor`, this will bounce out of the C back into Python. It will carry `self` over and any "simple" arguments. Can return scalars.
+
+_Scalar iniling_
+
+DaCe will optimize aas much as it can. This means any scalar with be turned into the value if they are never written to. This can lead to subtle bug because of our mode of distributed build. For example `da_min` is a grid variable which is decomposition dependant and read-only in the critical code path. This will be inlined, leading to error when executing larger decomposition (because it uses the `da_min` of the build decomposition). Our workaround for now is a callback (pending a series of fix from DaCe for a better solution).
+
+_Parsing errors_
+
+DaCe cannot parse _any_ dynamic Python and any code that allocates memory on the fly (think list creation). It will also complain about any arguments it can't memory describe (remember `dace_compiletime_args` )
```

Comment:
```suggestion
DaCe cannot parse _any_ dynamic Python and any code that allocates memory on the fly (think list creation). It will also complain about any arguments it can't memory describe (remember `dace_compiletime_args` ).
```